#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.279682401374145 0.6457002455977812 0.7105272318472786 1.0949142781468113
  position -1.366018158314381 -0.9122162696844922 1.2264968426628304
}
Background {
  skyColor [
    0.27451 0.27451 0.27451
  ]
}
TexturedBackgroundLight {
}
RectangleArena {
}
DEF Wheelie Robot {
  rotation 0.0118880680651626 0.008118639856034168 -0.9998963753932535 0.06030067395055892
  children [
    DEF BAR Group {
      children [
        DEF barattachL Solid {
          translation 0.03 -0.04 0.04
          children [
            DEF barattachL Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.03 0.01
              }
            }
          ]
          name "barattachL"
          boundingObject USE barattachL
          physics Physics {
          }
        }
        DEF barattachR Solid {
          translation -0.03 -0.04 0.04
          children [
            DEF barattachR Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.03 0.01
              }
            }
          ]
          name "barattachR"
          boundingObject USE barattachR
          physics Physics {
          }
        }
        DEF barattachForw Solid {
          translation 0 -0.07 0.04
          children [
            DEF barattachR Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.01 0.01
              }
            }
          ]
          name "barattachForw"
          boundingObject USE barattachR
          physics Physics {
          }
        }
        DEF Bar Solid {
          translation 0 -0.06 0.04
          children [
            DEF bar Shape {
              appearance PBRAppearance {
                baseColor 1 0.333333 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.1 0.01 0.015
              }
            }
          ]
          name "bar"
          boundingObject USE bar
          physics Physics {
          }
        }
      ]
    }
    DEF Body_traf Transform {
      translation 0 0 0.0425
      children [
        DEF Body Shape {
          appearance PBRAppearance {
            baseColor 0 0.333333 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 3.257266666490435
        anchor 0.045 0 0.025
      }
      device [
        RotationalMotor {
          name "motor1"
        }
        PositionSensor {
          name "ps1"
        }
      ]
      endPoint Solid {
        translation 0.04499989872758694 3.598920572876703e-07 0.025006239522075015
        rotation 0.7052787799078534 -0.023098944242319543 0.7085536545573525 3.1875829672510885
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0.666667 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF wheel_phys Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 3.2691729871404154
        anchor -0.045 0 0.025
      }
      device [
        RotationalMotor {
          name "motor2"
        }
        PositionSensor {
          name "ps2"
        }
      ]
      endPoint Solid {
        translation -0.04500014428240732 8.413296563137958e-07 0.02500568625336551
        rotation 0.7052036750553536 -0.027319521827373055 0.7084782427255948 3.1959798052422768
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics USE wheel_phys
      }
    }
    BallJoint {
      jointParameters BallJointParameters {
        position -10.910677804383637
        anchor 0 -0.035 0.005
      }
      endPoint Solid {
        translation 3.994844040766665e-05 -0.03581412742288446 0.005002325342375656
        rotation -0.5555079134160478 -0.7976203423623416 -0.23497392957928318 2.7737923614231845
        children [
          DEF Caster_wheel Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Sphere {
              radius 0.005
            }
          }
        ]
        name "solid(6)"
        boundingObject DEF Caster_wheel Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.005
          }
        }
        physics Physics {
        }
      }
    }
    BallJoint {
      jointParameters BallJointParameters {
        position -9.612106004685316
        anchor 0 0.035 0.005
      }
      endPoint Solid {
        translation -4.589707028679335e-05 0.03591285835089815 0.0050017019064049
        rotation -0.4696472990403759 0.7847221384601062 0.4045276009306131 2.863414466864181
        children [
          DEF Caster_wheel Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Sphere {
              radius 0.005
            }
          }
        ]
        name "solid(4)"
        boundingObject DEF Caster_wheel Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.005
          }
        }
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation -0.0065 0 0.0045
      rotation 0.5773489358556708 0.5773509358554485 -0.5773509358554485 2.0944
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.666667 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "dsr"
      type "infra-red"
    }
    DistanceSensor {
      translation -0.02 0 0.0045
      rotation 0.5773489358556708 0.5773509358554485 -0.5773509358554485 2.0944
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.666667 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "dsr2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.0125 0.0045
      rotation 0 1 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.666667 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "dsm"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.0065 0 0.0045
      rotation 0.5773489358556708 0.5773509358554485 -0.5773509358554485 2.0944
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsl"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.02 0 0.0045
      rotation 0.5773489358556708 0.5773509358554485 -0.5773509358554485 2.0944
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsl2"
      type "infra-red"
    }
    DistanceSensor {
      translation -0.0399996 -0.0253397 0.0725011
      rotation 2.0897299999898513e-06 -2.3122099999887704e-06 0.9999999999951434 -2.094395307179586
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf1"
      type "infra-red"
    }
    DistanceSensor {
      translation -0.0149997 -0.0425 0.0725003
      rotation 2.4206499999917205e-06 -9.90502999996612e-07 0.9999999999965796 -1.8325953071795862
      children [
        Solid {
          rotation 0 1 0 -1.5707953071795862
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.0150003 -0.0425 0.0725
      rotation -1.4877799999853769e-06 4.176629999958949e-06 -0.9999999999901712 1.309
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf3"
    }
    DistanceSensor {
      translation 0.0399996 -0.0253397 0.0725011
      rotation -1.1321599999835046e-06 5.2780599999231e-06 -0.9999999999854301 1.0472
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf4"
    }
    DistanceSensor {
      translation 0.0433202 -0.0153397 0.0725012
      rotation 0.27974010365477736 0.8664353210485701 -0.41356415324188295 5.92879e-06
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf5"
    }
    DistanceSensor {
      translation 0.0433202 0.0146603 0.0725012
      rotation 5.663319999692065e-06 8.756389999523883e-06 0.9999999999456264 0.523596
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf6"
    }
    DistanceSensor {
      translation 0 0.0485062 0.0725012
      rotation 3.3977199999752487e-06 1.7391999999873305e-06 0.9999999999927154 1.57079
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf7"
    }
    DistanceSensor {
      translation -0.0433202 0.0146603 0.0725012
      rotation 2.790649999989106e-06 -1.410429999994494e-07 0.9999999999960962 2.61799
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf8"
    }
    DistanceSensor {
      translation -0.0433202 -0.0153397 0.0725012
      rotation 2.568459999990645e-06 -8.292599999969797e-07 0.9999999999963578 3.14159
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.005 0.005
              }
            }
          ]
        }
      ]
      name "dsf9"
    }
  ]
  boundingObject USE Body_traf
  physics Physics {
  }
  controller "turnforgiventime"
}
